cylc/flow/scheduler_cli.py

```diff
@@ -417,35 +365,9 @@ async def _run(parser, options, reg, is_restart, scheduler):
         return ret
 
 
-def main(*args, is_restart=False):
-    """Abstraction for cylc (run|restart) CLI"""
-    # the cli_function decorator changes the function signature which
-    # irritates pylint.
-    if is_restart:
-        return restart(*args)  # pylint: disable=E1120
-    else:
-        return run(*args)  # pylint: disable=E1120
-
-
-@cli_function(partial(get_option_parser, is_restart=True))
-def restart(parser, options, *args):
-    """Implement cylc restart."""
-    return scheduler_cli(parser, options, args, is_restart=True)
-
-
-@cli_function(partial(get_option_parser, is_restart=False))
-def run(parser, options, *args):
-    """Implement cylc run."""
+@cli_function(get_option_parser)
+def play(parser, options, *args):
+    """Implement cylc play."""
     if not args:
         _auto_install()
```

Comment:
Can kill this and the `_auto_install` bit now the optional `REG` is no longer optional.

Reply:
```diff
diff --git a/cylc/flow/scheduler_cli.py b/cylc/flow/scheduler_cli.py
index 2a594e9d1..484940b54 100644
--- a/cylc/flow/scheduler_cli.py
+++ b/cylc/flow/scheduler_cli.py
@@ -207,19 +207,6 @@ RunOptions = Options(
     get_option_parser(add_std_opts=True), DEFAULT_OPTS)
 
 
-def _auto_install():
-    """Register workflow installed in the cylc-run directory"""
-    try:
-        reg = suite_files.register()
-    except SuiteServiceFileError as exc:
-        sys.exit(exc)
-    # Replace this process with "cylc play REG ..." for 'ps -f'.
-    os.execv(
-        sys.argv[0],
-        [sys.argv[0]] + sys.argv[1:] + [reg]
-    )
-
-
 def _open_logs(reg, no_detach):
     """Open Cylc log handlers for a flow run."""
     if not no_detach:
```